---
title: "ANGSD - HW Week 3"
author: "Tomer M. Yaron"
date: "1/27/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown - HW 2 - Tomer M. Yaron

## 1 - Getting the fastq files

Logging in to the server:

```{cars}
ssh toy2005@aristotle.med.cornell.edu
ssh buddy.pbtech
```

Create a directory for the HW:

```{cars}
cd angsd
mkdir HW2
cd HW2
mkdir databases
cd databases
mkdir ENA_files
```

### a - Getting the TEXT and mapping files:

```{cars}
wget --content-disposition 'https://www.ebi.ac.uk/ena/data/warehouse/filereport?accession=PRJEB5348&result=read_run&fields=study_accession,sample_accession,secondary_sample_accession,experiment_accession,run_accession,tax_id,scientific_name,instrument_model,library_layout,fastq_ftp,fastq_galaxy,submitted_ftp,submitted_galaxy,sra_ftp,sra_galaxy,cram_index_ftp,cram_index_galaxy&download=txt'

wget 'https://ndownloader.figshare.com/files/2194841' -O ERP004763_sample_mapping.tsv
```

### b - Obtaining the ENA names for WT biological replicates no. 2:

```{cars}
grep -P 'WT\t2$' ERP004763_sample_mapping.tsv | cat > bio_rep_2.txt
cat bio_rep_2.txt
```

ERR458878	1	WT	2  
ERR458879	2	WT	2  
ERR458880	3	WT	2  
ERR458881	4	WT	2  
ERR458882	5	WT	2  
ERR458883	6	WT	2  
ERR458884	7	WT	2  

______________________________________________________________________

### c - Downloading the FASTQ file for WT biological replicates no. 2:

```{cars}
for line in $(awk '{print $1}' bio_rep_2.txt); do
  egrep $line PRJEB5348.txt | wget -P ./ENA_files "ftp://$(cut -f10)";
done
```

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/ERR458878/ERR458878.fastq.gz
           => ‘ERR458878.fastq.gz’  
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/ERR458879/ERR458879.fastq.gz
           => ‘ERR458879.fastq.gz’  
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/ERR458880/ERR458880.fastq.gz
           => ‘ERR458880.fastq.gz’  
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/ERR458881/ERR458881.fastq.gz
           => ‘ERR458881.fastq.gz’  
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/ERR458882/ERR458882.fastq.gz
           => ‘ERR458882.fastq.gz’  
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/ERR458883/ERR458883.fastq.gz
           => ‘ERR458883.fastq.gz’  
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/ERR458884/ERR458884.fastq.gz
           => ‘ERR458884.fastq.gz’  

______________________________________________________________________

## 2 - Multiple FASTQ files per sample

In the experiment they ran, they did 48 biological replicates and *7 technical replicates*. Therefore, for each biological replicate there are 7 differecnt FASTQ files, one per technical replicate (7 different lanes of a single flow-cell on an Illumina HiSeq 2000).

_____________________________________________________________________

## 3 - Number of reads

```{cars}
for f in ./*; do
  echo Run Accession $(echo $f | cut -d'.' -f2 | cut -d'/' -f2): $(zcat $f | egrep '^@ERR' -c);
done

```

Run Accession ERR458878: 1467569  
Run Accession ERR458879: 1450012  
Run Accession ERR458880: 1441569  
Run Accession ERR458881: 1321748  
Run Accession ERR458882: 1131784  
Run Accession ERR458883: 1140688  
Run Accession ERR458884: 1461646  

_____________________________________________________________________

Writing the results into a file:

```{cars}

echo $(for f in ./*; do echo Run Accession $(echo $f | cut -d'.' -f2 | cut -d'/' -f2): $(zcat $f | egrep '^@ERR' -c)\;; done) > ./reads_number.txt #For writing the results into a file
cat reads_number.txt

```

Run Accession ERR458878: 1467569; Run Accession ERR458879: 1450012; Run Accession ERR458880: 1441569; Run Accession ERR458881: 1321748; Run Accession ERR458882: 1131784; Run Accession ERR458883: 1140688; Run Accession ERR458884: 1461646;

_____________________________________________________________________

## 4 - Biological project

On October 2018 a paper was publiched in Nature regarding single-cell transcriptomics of 20 mouse organs. I am interested in analyzing this data from the aspect of kinases in defferent tissues, and to compare it to RNAseq in human cancer cells from TCGA.













