---
title: "ANGSD - HW Week 5"
author: "Tomer M. Yaron"
date: "2/12/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown - HW 4 - Tomer M. Yaron

## 1 - A script that runs BWA and STAR (using the indexed genome)

```{BWA_STAR}
#!/bin/bash

spack load bwa@0.7.15%gcc@6.3.0
spack load star@2.6.1a

bwa mem ./../databases/referenceGenomes/sacCer3_BWAindex/sacCer3 ./../databases/ENA_files/ERR458878.fastq.gz > /athena/elementolab/scratch/toy2005/angsd/hw4/BWA/ERR458878.bwa.sam
STAR --runMode alignReads --runThreadN 8 --genomeDir ./../databases/referenceGenomes/sacCer3_STARindex/ --readFilesIn ./../databases/ENA_files/ERR458878.fastq.gz --readFilesCommand zcat --outFileNamePrefix /athena/elementolab/scratch/toy2005/angsd/hw4/STAR/ERR458878. --outSAMtype BAM SortedByCoordinate
```

_____________________________________________________________________

## 2 - Create a subset of the aligned reads that map to chromosome 1

```{CHR1}
samtools view -h ERR458878.bwa.bam | awk '$3 == "chrI"' > ERR458878.bwa.chrI.bam
samtools view -h ERR458878.Aligned.sortedByCoord.out.bam | awk '$3 == "chrI"' > ERR458878.Aligned.chrI.sortedByCoord.out.bam
```

_____________________________________________________________________

## 3 - Compring the differences between BWA and STAR

### a - STAR optional fields

STAR shows 4 optional fields: NH, HI, AS, nM.

1. NH: i == integer. The number of reported alignments that match this query, i.e. the number of different pocations to which this read was matched.
2. HI: i == integer. Hit index - i.e. the place in the SAM/BAM file in which this read appears. In the case that there are multiple records for the same read (or its mate), then the HI is the way to distinguish between them.
3. AS: i == integer. The alignment score generated by aligner, which depends on the matches, and the penalties for the gap and mismatches.
4. nM: i == integer. Number of mismatches per paired alignment.

### b - BWA optional fields

BWA shows 4 optional fields: NH, HI, AS, nM.

1. NM: i == integer. The number of edits from the read to the reference. Practically - the number of mismatched including N-base.
2. MD: Z == string. A string contains the mismatches positions (complementary to the CIGAR string).
3. AS: i == integer. The alignment score generated by aligner, which depends on the matches, and the penalties for the gap and mismatches.
4. XS: i == integer. the score of the 2nd best alignment.
5. XA (sometimes): Z == string. Alternative hits in the format: <chr,pos,CIGAR,NM;>

### c - Interpretation of the mapping quality

In STAR, the mapping quality is calculated similar to TopHat, except that the value for uniquely mapped reads is 255 instead of 50:  
* 0 = maps to 5 or more locations  
* 1 = maps to 3-4 locations  
* 3 = maps to 2 locations  
* 255 = unique mapping  

In BWA, the mapping quality is the actual Phred scores. The calculation is based on the number of optimal (best) alignments found, as well as the number of sub-optimal alignments combined with the Phred scores of the bases which differ between the optimal and sub-optimal alignments. It equals -10 log10 {Prfmapping position is wrong}, rounded to the nearest integer. A value 255 indicates that the mapping quality is not available.

### d - Splitted reads

In order to find the splitted read - which are splice junctions, we will look at ERR458878.SJ.out.tab file. The columns in the file have the following meaning:  
* column 1: chromosome  
* column 2: first base of the intron (1-based)  
* column 3: last base of the intron (1-based)  
* column 4: strand (0: undefined, 1: +, 2: -)  
* column 5: intron motif: 0: non-canonical; 1: GT/AG, 2: CT/AC, 3: GC/AG, 4: CT/GC, 5: AT/AC, 6: GT/AT  
* column 6: 0: unannotated, 1: annotated (only if splice junctions database is used)  
* column 7: number of uniquely mapping reads crossing the junction  
* column 8: number of multi-mapping reads crossing the junction  
* column 9: maximum spliced alignment overhang  

In order to find splitted reads, we will look for read which their CIGAR string contains N (uskip region == gap). for that we will use the following command:

```{STAR_SPLIT}
awk '/[0-9]N/' ./ERR458878.Aligned.chrI.sortedByCoord.out.sorted.bam | head
```

we got:

```
ERR458878.1021866	272	chrI	140241	1	43M566N8M	*	0	0	GGATAATGGAGCGACATCCAACAACAATAGATCTTGAGTCTTGAATCAAGA	?4>B9D<FD?8:9BGF>CA823<,HH@HCEB<C;FE:;DDCA222244111	NH:i:4	HI:i:2	AS:i:41	nM:i:0  
ERR458878.1055117	272	chrI	7860	3	11M125565N40M	*	0	0	TTCCAACAGAAATACACAAAAGAAAGAAGATAATCAAGTGGTACCGAAAAG	GBGJJJJJGIHGDGHHGIIGG@JJJIIGGGHIIGJJIGFFFBCFFDBD@@@	NH:i:2	HI:i:2	AS:i:39	nM:i:0  
ERR458878.106011	256	chrI	73033	3	39M144745N12M	*	0	0	AAACTGTCCAATCATCTTGGTTACCAGATGCCCAAGAGCTTTTAATTTTTT	=??1DD,?DDD>??DEEDFE<CCFEEDFCF<F1?#################	NH:i:2	HI:i:2	AS:i:39	nM:i:0  
ERR458878.1066648	16	chrI	142248	255	6M366N45M	*	0	0	TGAAGGTACTGCTGTTTCTCAAGCTGACGTCACTGTCTTCAAGGCTTTCCA	@D??<FB08GF@BBB??BC?BGB@;C2AA333BFA<A?BCBDA:?DDD7@@	NH:i:1	HI:i:1	AS:i:51	nM:i:0  
ERR458878.1069019	256	chrI	81343	3	11M61419N40M	*	0	0	AGAAGAAGAGGTGACGATGTCGATTTATTCGGTTCCGACGATGAAGAAGCT	@<?BDD:BHDD<AFGHE??E?BDGIIICA?31CD;@@GIBACFF;@FFFH@	NH:i:2	HI:i:2	AS:i:39	nM:i:0  
ERR458878.1083582	0	chrI	142219	255	35M366N16M	*	0	0	AACGCTTCTTTGGCTGACAAGTCATACATTGAAGGTACTGCTGTTTCTCAA	CC@FFFFFHHHHHJJJJJJJJCHIIGJGIJJCHIJDEHIJJJJIJJJIJFJ	NH:i:1	HI:i:1	AS:i:51	nM:i:0  
ERR458878.1085919	256	chrI	72727	3	43M91528N8M	*	0	0	AGTTTCCGATGTCGGTAACGCTCTCTTGGATGGTGCTGACTGTGCTATGTT	:;?D7D4A82<DA=AEEE+1C:)::?*:C*:3@*:4?*9)**09*9?B=B#	NH:i:2	HI:i:2	AS:i:45	nM:i:1  
ERR458878.1086228	16	chrI	94493	3	37M33658N14M	*	0	0	GTATTTGTATATCAAAAAATGATCCTGTGATTTTTTCAAAAAGAAAAAAAA	HHHHHHJIJGHEGJJJJJIJJIIJJJJJIIJJJJJIIJHHHHHFFFFFBCC	NH:i:2	HI:i:1	AS:i:39	nM:i:0  
ERR458878.1086228	272	chrI	94493	3	37M97949N14M	*	0	0	GTATTTGTATATCAAAAAATGATCCTGTGATTTTTTCAAAAAGAAAAAAAA	HHHHHHJIJGHEGJJJJJIJJIIJJJJJIIJJJJJIIJHHHHHFFFFFBCC	NH:i:2	HI:i:2	AS:i:39	nM:i:0  
ERR458878.1090280	256	chrI	119210	1	12M85128N38M1S	*	0	0	CAATACACCAACAACTGCTAGCACCATCATAACTACGACCGAACCATGGAC	@@@+BDDAFFF=B?:<<ECH><>F=;FFCFEGF4C<;0DFFE@@FF;=@##	NH:i:3	HI:i:3	AS:i:36	nM:i:5
```

Let's take a look on read ERR458878.1055117:

```
ERR458878.1055117	272	chrI	7860	3	11M125565N40M	*	0	0	TTCCAACAGAAATACACAAAAGAAAGAAGATAATCAAGTGGTACCGAAAAG	GBGJJJJJGIHGDGHHGIIGG@JJJIIGGGHIIGJJIGFFFBCFFDBD@@@	NH:i:2	HI:i:2	AS:i:39	nM:i:0
```

Its CIGAR string is 11M125565N40M, which means we have 11 matches, then 125565 skips (gaps), then 40 matches again (51 matches in total). STAR identified this read as mapped to two sides of a splice junction.

However, in BWA, this is how the read is presented:

```{BWA_SPLIT}
egrep 'ERR458878.1055117' ERR458878.bwa.chrI.sorted.bam
```

which results in:

```
ERR458878.1055117	16	chrI	133436	60	11S40M	*	0	0	TTCCAACAGAAATACACAAAAGAAAGAAGATAATCAAGTGGTACCGAAAAG	GBGJJJJJGIHGDGHHGIIGG@JJJIIGGGHIIGJJIGFFFBCFFDBD@@@	NM:i:0	MD:Z:40	AS:i:40	XS:i:0
```

As we can see, the CIGAR string for this read in BWA is 11S40M, which means 11 mismatches and then 40 matches. Where STAR handle this read with a big gap and therefore identified this site as a splice junction, BWA did not take into account a splice junction and therefore consider the 11 first bases as mismatch to the next 40 bases, instead of as to matches with an intron in between.



