---
title: "ANGSD - HW Week 10"
author: "Tomer M. Yaron"
date: "3/27/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown - HW 10 - Tomer M. Yaron

## 1 - Analyze scRNA-seq

```{r, message=F, warning=F, read_files}

library(SingleCellExperiment)
library(DropletUtils)
library(scater)
library(ggplot2)
library(EnsDb.Hsapiens.v86)
library(scran)
library(Seurat)
library(GGally)

count_matrix <- read.table('databases/WT-1.dge.txt', sep = '\t', header = TRUE, row.names = 1)
dim(count_matrix)

```

### a - Data explanation

In the count matrix, The rows represent different genes and the coluns represent different cell, marked by their specific barcode (whereas every entry is the gene count for that specific cell/barcode). In this assay, 1,400 cells were characterized, along 25,319 different genes.

### b - SingleCellExperiment object

```{r, message=F, warning=F, sce}

sc_exp <- SingleCellExperiment(assays = list(counts = as.matrix(count_matrix)))

```

### c - Accessing counts (two ways)

```{r, message=F, warning=F, counts}

assay(sc_exp,"counts")[1:10,1:10]
counts(sc_exp)[1:10,1:10]

```

### d - Calculating sequence depth

```{r, message=F, warning=F, reads}

colSums(counts(sc_exp)[,1:5])

```

### e - Expressed genes

Here I calculated the number of genes with non-zero values across **all** the 5 cells, i.e. genes which are expressed in each one of the first 5 cells.

```{r, message=F, warning=F, expressed_genes}

sum(rowSums(counts(sc_exp)[,1:5] != 0) == 5)

```

### f - Keep track of the original rows and columns names

```{r, message=F, warning=F, row_col_names}

org_row_names <- rownames(sc_exp)
org_col_names <- colnames(sc_exp)

```

### g - Keep track of the original rows and columns names

We will use the libraries _DropletUtils_ and _scater_, sepcifically the functions _barcodeRanks_ and _calculateQCMetrics_, where the features _$total_ and _$total\_counts_ of those objects respectively, represents the total number of reads (==UMI) which we calculated in section d. In addition, the feature _total\_features\_by\_counts_ in the _calculateQCMetrics_ object represents the number of expressed genes (genes with a non-zero read counts, i.e. barcode detection).

```{r, message=F, warning=F, histogram}

chr_loc <- mapIds(EnsDb.Hsapiens.v86, keys=rownames(sc_exp), column="SEQNAME", keytype="GENENAME")

bcrank <- barcodeRanks(counts(sc_exp))
sc_exp <- calculateQCMetrics(sc_exp, feature_controls=list(Mito=which(chr_loc=="MT")))

total_umi_exp_genes <- data.frame(sc_exp$total_counts,sc_exp$total_features_by_counts,sc_exp$pct_counts_Mito)
colnames(total_umi_exp_genes) <- c('UMI','GENES','MT')
class(total_umi_exp_genes)
ggplot(total_umi_exp_genes, aes(x=UMI)) + geom_histogram(bins = 100) + ggtitle("Total UMI Counts") + ylab("Cells") + theme(plot.title = element_text(hjust = 0.5))
ggplot(total_umi_exp_genes, aes(x=GENES)) + geom_histogram(bins = 100) + ggtitle("Total Expressed Genes") + ylab("Cells") + theme(plot.title = element_text(hjust = 0.5))

```

The first histogram (Total UMI) represents the total number of detected barcodes per cell. In equivalence to bulk RNA-seq, this is the "number of reads".

The second histogram is a binary result, whether the gene is expressed or not in the cell. It should be noted that this is **not** the equivalent to the normalized expression in bulk RNA-seq, since it is only a binary answer whether the gene was expressed (its barcode was detected at least one time) or not.

From the histograms, we can see that out of 1400 cells, msot of the cells has sequencing depth of less than 10K, and between 100-4000 genes detected out of 25,319 measured genes. To be noted that since the RNA expression levels in single cell are very low, a lot of genes might not be detected due to noise, or due to other biological reasons (cell cycle etc.).

```{r, message=F, warning=F, histogram_mt}

ggplot(total_umi_exp_genes, aes(x=MT)) + geom_histogram(bins = 100) + ggtitle("Mitochondrial Genes") + xlab("% Mitochondrial Genes Reads") + ylab("Cells") + theme(plot.title = element_text(hjust = 0.5))

```

As we can see in the histogram, most of the cells have less than 10% mitochondrial reads.

### h - QC filtration

There are several different ways to filter cell with low quality, some of the ways are more stringent and some are less harsh. Usually, we would like to remove cells with low reads counts or total number of expressed genes, assuming that those cells suffer from high noise and/or technical issues. However, this would probably remove cells with low RNA content, especially cell types with low RNA content in general, and we can see that the vast majoritiy of the cells in this database has relatively low amount of reads. Therefore, I would use a softer filter of removing cells with high proportion of mitochondrial RNA.

```{r, message=F, warning=F, qc_filter}

high_mito <- isOutlier(sc_exp$pct_counts_Mito, nmads=3, type="higher") | is.na(sc_exp$pct_counts_Mito)
sc_exp_filter <- sc_exp[,!high_mito]
summary(high_mito)

```

We filtered out 326 cells out of 1400.

### i - Normalization

```{r, message=F, warning=F, normalization}

set.seed(1000)
clusters <- quickCluster(sc_exp_filter, method="igraph", min.mean=0.1, irlba.args=list(maxit=1000))
table(clusters)

sc_exp_filter <- computeSumFactors(sc_exp_filter, min.mean=0.1, cluster=clusters)
summary(sizeFactors(sc_exp_filter))

plot(sc_exp_filter$total_counts, sizeFactors(sc_exp_filter), log="xy")

```

The size factor calculated show the expected behavior (linear).

We will use the function _normalize_ for normalizing the data (with the option _return\_log = FALSE_ in order to get the log-normalized and the normalized), and then the option _normcounts_ for accessing the normalized counts matrix:

```{r, message=F, warning=F, norm_matrix}

sc_exp_filter <- normalizeSCE(sc_exp_filter)
sc_exp_filter_norm_matrix <- logcounts(sc_exp_filter)
head(sc_exp_filter_norm_matrix[,1:5], 10)

```

______________________________________________________________________

## 2 - Seurat

### a - Seurat object

```{r, message=F, warning=F, seurat}

sce_srt <- CreateSeuratObject(count_matrix, min.cells = -1, min.genes = -1)

```

There are two cells with not expression at all ("GCCCGTCTATGT" & "GACAAACCTACC"), and therefore when we creat a Seurat object, with the default parameter for _min.genes = 0_, we filter out those two cells (because it looks for colSums > min.genes). In order to include those cells, we have to set the parameters to be -1.

### b - Filtering

We will find all the mitochondrial genes (start with "MT"), and will manually calculate the percentage of mitochondrial RNA.

```{r, message=F, warning=F, srt_filter}

mito_genes <- rownames(sce_srt@data)[which(chr_loc == 'MT')]
percent_mito <- colSums(sce_srt@raw.data[mito_genes, ]) / colSums(sce_srt@raw.data)

high_mito_srt <- isOutlier(percent_mito, nmads=3, type="higher") | is.na(percent_mito)
sce_srt_filter <- SubsetData(sce_srt, cells.use = names(high_mito_srt[!high_mito_srt]))
summary(high_mito_srt)

```

As in Question 1, we filtered out 326 cells with an outlier percentage of expressed mitochondrial genes.

### C - Normalization

We will normalize the data using _NormalizeData_ (LogNormalize method), and then access the normalized matrix using _Object@data_.

```{r, message=F, warning=F, srt_norm}

sce_srt_filter <- NormalizeData(sce_srt_filter)
sce_srt_filter_norm_matrix <- sce_srt_filter@data
head(sce_srt_filter_norm_matrix[,1:5], 10)

```

### d - Compare the first 10 cells

We will exclude the genes which have zero expression across all 10 samples:

```{r, message=F, warning=F, comparison}

sce_comp_10 <- sc_exp_filter_norm_matrix[,1:10]
sce_srt_comp_10 <- as.matrix(sce_srt_filter_norm_matrix[,1:10])

sce_comp_10 <- sce_comp_10[which(rowSums(sce_comp_10) != 0),]
sce_srt_comp_10 <- sce_srt_comp_10[which(rowSums(sce_srt_comp_10) != 0),]

dim(sce_comp_10)
dim(sce_srt_comp_10)

```

We can see that in both cases we filtered out 11078 genes which are not expressed at all across the first 10 samples, and we have 14241 genes left.

Now we will plot the pairwise comparison using _ggduo_ from the _GGally_ package:

```{r, message=F, warning=F, comparison_plot}

exp_comb_matrix <- data.frame(cbind(sce_comp_10,sce_srt_comp_10))
ggduo(exp_comb_matrix, columnsX = 1:10, columnsY = 11:20)

```

As we can see, there is a very strong correlation between the normalized expression of similar cells in SingleCellExperiment object and the Seurat object, whereas for different cells, there is a correlation but not as strong. In addition, to be noted that the actual values of the normazlied expression can be a little bit different between the two objects, since both objects are using a slightly different normalization method (SCE is using the size factor whereas Seurat is using simply the log-normalization).

______________________________________________________________________

## 3 - Cell type

According to the pairwise graph from Question 2 (assuming the first 10 cells are a good representation of the entire population), we can conclude that the cells are fairly homogeneous,but still have some different between each other in the expression of different genes. In particular, we can see that there are some genes whih are not expressed at all in one cell but are expressed in the other cell and vice versa (those are the dots along the X and Y axes). However, most of the highly expressed genes are expressed in all the cells, suggesting that the cells are from the same type.

In conclusion, I would guess that those cells are **immune cells**. This cell population is fairly homogeneous relatively to the high expressed genes which make those cells to be immune cells, but still has significant difference between different types of cells inside this population (CD4, CD8, macrophages etc.). The high correlation between the highly expressed genes on the one hand, and the genes which are expressed only in certain cells on the other, make me believe that those cells are different types of immune cells.


